lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,213 @@
+                error = "The link you tried to click expired or was revoked. Please try again.";
+            }
+        }
+        this.state = {
+            loaded: false,
+            loginAttemptInfo: undefined,
+            error,
+        };
+    }
+
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    getModifiedRecipeImplementation = (): RecipeInterface => {
+        return {
+            ...this.props.recipe.recipeImpl,
+            createCode: async (input) => {
+                let contactInfo;
+                if ("email" in input) {
+                    contactInfo = input.email;
+                } else {
+                    contactInfo = formatPhoneNumberIntl(input.phoneNumber);
+                }
+                const res = await this.props.recipe.recipeImpl.createCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...res,
+                        lastResend: new Date().getTime(),
+                        contactInfoType: this.props.recipe.config.contactInfoType,
+                        contactInfo,
+                    };
+                    this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                }
+                return res;
+            },
+            resendCode: async (input) => {
+                if (!this.state.loginAttemptInfo) {
+                    // This should never happen, but it makes TS happy
+                    throw new Error("Resend without loginAttemptInfo");
+                }
+                const res = await this.props.recipe.recipeImpl.resendCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...this.state.loginAttemptInfo,
+                        lastResend: new Date().getTime(),
+                    };
+                    this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+
+                    this.setState((os) => ({
+                        ...os,
+                        error: "Please try again.",
+                        loginAttemptInfo: undefined,
+                    }));
+                } else if (res.status === "GENERAL_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
```

Comment:
Probably not a good idea to clear the login attempt info here. Maybe the general error is just some custom error by the developer